backend/corpora/lambdas/api/v1/curation/collections/common.py

```diff
@@ -0,0 +1,121 @@
+from ...authorization import is_user_owner_or_allowed
+from ......common.corpora_config import CorporaConfig
+from ......common.corpora_orm import (
+    CollectionVisibility,
+    DbCollectionLink,
+    DbCollection,
+    DbDataset,
+    DbDatasetProcessingStatus,
+    DbDatasetArtifact,
+)
+
+
+def reshape_for_curation_api_and_is_allowed(collection, token_info, allow_access=False):
```

Comment:
that said, I also still think it might be a cleaner pattern to extract the access_type bit into a helper function, do the boolean eval in the handlers that need it, and do only in-place modification in the reshape function (and thus get rid of the flag altogether).

Reply:
@nayib-jose-gloria in a separate branch, if you'd like to demonstrate how that'd look, I'd be happy to review and give my opinion.